File: apps/backend/src/domain/entity/list/list-settings.entity.ts
import {ListSettingsRepository} from "../../repository/list/list-settings.repository";
import {Entity, PrimaryKey, Property} from "@mikro-orm/core";
import { createId } from "../../util/create-id.fn";

@Entity({
	tableName: "list_settings",
	repository: () => ListSettingsRepository,
})
export class ListSettingsEntity
{
	@PrimaryKey({ type: 'varchar', length: 36, })
	public id: string = createId();

	@Property()
	public stackSize!: number;

	@Property()
	public allowChunkMerge!: boolean;

	@Property()
	public maxChunkLength!: number;

	@Property()
	public maxChunkJoinLength!: number;
}

File: anistats_app/data_access/lib/src/entity/list_settings_entity.dart
import 'package:copy_with_extension/copy_with_extension.dart';

part 'list_settings_entity.g.dart';

@CopyWith()
class ListSettingsEntity {
  const ListSettingsEntity({
    required this.allowChunkMerge,
    required this.maxChunkLength,
    required this.maxChunkJoinLength,
    required this.stackSize,
  });

  final bool allowChunkMerge;
  final Duration maxChunkLength;
  final Duration maxChunkJoinLength;
  final int stackSize;
}
